content/en/docs/How Tos/metrics.md

```diff
@@ -134,13 +73,161 @@ or an error saying why it can't.
+    - `NumFilteredLines`, the number of discarded (i.e., filtered) records
+- [Output](https://pkg.go.dev/github.com/AdRoll/baker#OutputStats):
+    - `NumProcessedLines`, the total number of processed records since the component creation
+    - `NumErrorLines`, the number of records that have produced an error
+- [Upload](https://pkg.go.dev/github.com/AdRoll/baker#UploadStats):
+    - `NumProcessedFiles`, the total number of processed files since the component creation
+    - `NumErrorFiles`, the number of files that have produced an error
+
+Due to historical reasons, these fields have the word _lines_ in them but they do 
+mean the number of records.
+
+The code example of 
+[how-to-create-filter](https://getbaker.io/docs/how-tos/create_filter/#processing-records) 
+shows how to correctly reports metrics in a custom Filter.
+
+### Report custom metrics
+
+Components may need to report custom metrics for monitoring some specific events. 
+Baker supports two main ways to expose custom metrics, namely:
+
+- return a 
+[`baker.MetricsBag`](https://pkg.go.dev/github.com/AdRoll/baker#MetricsBag)
+instance from the `Stats` method 
+- directly use the 
+[`baker.MetricsClient`](https://pkg.go.dev/github.com/AdRoll/baker#MetricsClient) in 
+the component code
+
+The two mechanisms following pull vs push approach respectively.
+The `MetricsBag` should be returned by the `Stats` method along with the default
+statistics and it will be collected 1 per second by the Baker runtime. Differently,
+the `MetricsClient` can be requested from the Baker topology in the component
+instantiation and it can be used in any part of the component code to report metrics.
+If there is no particular requirements it is suggested to prefer the `MetricsBag`. 
+Indeed, the pull approach permits the reduction the metrics overhead during the 
+record processing.
+
+Both `MetricsBag` and `MetricsClient` Metrics supports the most common metric types,
+namely:
+- `RawCounter`, a cumulative counter that can only increase.
+- `DeltaCounter`, the total number of event occurrences in a unit time.
+- `Gauge`, a snapshot of an event in the last unit time. 
+- `Histogram`, a statistical distribution of a set of values in one unit 
+of time.
+- `Duration` or `Timing`, like an histogram but with time durations
+
+Moreover, the `MetricsClient` is the direct interface to the remote metrics service 
+used by the topology, e.g., *Datadog*, thus it supports some specific features such 
+as **tags**.
+Tags are a way of adding dimensions to telemetries so they can be filtered, 
+aggregated, and compared in different visualizations.
+Therefore, if a component requires to publish its metrics with a set of specific tags, the 
+`MetricsClient` should use rather than `MetricsBag`.
+However, it is suggested to use `MetricsClient` in the `Stats` method of the 
+components to reduce contention and improve performance in the processing of the 
+elements.
```

Comment:
together with the above sentence, the whole concept seems a bit fuzzy, I'd maybe rephrase a bit